### PR TITLE
DOC: added note that answer tests are currently run with nose

### PR DESCRIPTION
**Pull request by nastasha-w**
## PR Summary

I added a note to the development -> testing page of the docs, [Answer Testing](https://yt-project.org/docs/dev/developing/testing.html#answer-testing) section, indicating that the current instructions do not reflect current practice, as described [here](https://github.com/yt-project/yt/pull/4939#issuecomment-2241559129) by @neutrinoceros. This will hopefully save people some time when working on pull requests. 

I added this text to the very beginning of the 'Answer Tests' section, in a note (`.. note::`):
    "This section documents answer tests run with `pytest`. The plan is to switch to using `pytest` for answer tests at some point in the future, but currently (July 2024), answer tests are actually run using `nose`."

### CODE REVIEW

File: doc/source/developing/testing.rst [hunk 1]
```diff
@@ -152,6 +152,10 @@ More pytest options can be found by using the ``--help`` flag
 
 Answer Testing
 --------------
+.. note::
+    This section documents answer tests run with ``pytest``. The plan is to
+    switch to using ``pytest`` for answer tests at some point in the future,
+    but currently (July 2024), answer tests are actually run using ``nose``.
```
Comment (neutrinoceros) [inline doc/source/developing/testing.rst:158]:
```suggestion
    but currently (July 2024), answer tests are still implemented and run with ``nose``.
```

**Review by neutrinoceros — DISMISSED** [review]

File: doc/source/developing/testing.rst [hunk 2]
```diff
@@ -152,6 +152,10 @@ More pytest options can be found by using the ``--help`` flag
 
 Answer Testing
 --------------
+.. note::
+    This section documents answer tests run with ``pytest``. The plan is to
+    switch to using ``pytest`` for answer tests at some point in the future,
+    but currently (July 2024), answer tests are still implemented and run with ``nose``.
```
Comment (chrishavlin) [inline doc/source/developing/testing.rst:158]: also a question for @neutrinoceros but would it be helpful to link to an old version of the docs where the nose testing is described. e.g., https://yt-project.org/docs/4.0.0/developing/testing.html#answer-testing ? I'd rather not encourage folks to write new tests for nose if they can avoid it, but there may be situations where unfortunately it is required (e.g., writing a new frontend).

Comment (neutrinoceros) [reply doc/source/developing/testing.rst:158]: that's a great idea. I didn't know that old docs were still accessible !

Comment (chrishavlin) [reply doc/source/developing/testing.rst:158]: yup! and I have on occasion gone back to look at the nose testing docs when i find myself needing to run nosetests locally. So in that case, @nastasha-w , mind adding the following (or something similar)?

```suggestion
    but currently (July 2024), answer tests are still implemented and run with ``nose``.
    If needed, you can refer to the old documentation on answer testing `here <https://yt-project.org/docs/4.0.0/developing/testing.html#answer-testing>`_.
```

Comment (nastasha-w) [reply doc/source/developing/testing.rst:158]: seems reasonable. That is basically the situation in PR #4939, where changes to the projection backend are causing image comparison tests of the frontends to fail

Comment (nastasha-w) [reply doc/source/developing/testing.rst]: I've added some extra text with the link. Sorry it took a while; I forgot to set the environment variable for a quick docs build

**Comment by nastasha-w** [comment]
pre-commit.ci autofix

**Comment by nastasha-w** [comment]
I think I've got the .DS_Store files out if my histories now (here and on PR #4939 and PR #4955)

File: doc/source/cookbook/circle.reg [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# Region file format: DS9 version 4.1
```
Comment (chrishavlin) [inline doc/source/cookbook/circle.reg:1]: Should this file be included here? Looks like a file that might get generated when the docs build?

**Review by chrishavlin — DISMISSED** [review]
Thanks for adding the note and link to the old docs!

File: doc/source/cookbook/circle.reg (hunk 3 above)
Comment (nastasha-w) [reply doc/source/cookbook/circle.reg:1]: Oh yeah, I think it is actually. Ugh. I'll remove it.

**Comment by nastasha-w** [comment]
@chrishavlin removing the file also seems to have removed your review, but adding `circle.reg` is out of the overall pull request and the change logs now. (I've been getting a lot of use out of `rebase` these past few days.)

File: doc/source/cookbook/circle.reg (hunk 3 above)
Comment (nastasha-w) [reply doc/source/cookbook/circle.reg]: removed

**Review by neutrinoceros — APPROVED** [review]

**Pull request merged**